src/arviz_base/io_numpyro.py

```diff
@@ -538,3 +620,105 @@ def from_numpyro(
+    constant_data=None,
+    predictions_constant_data=None,
+    log_likelihood=None,
+    index_origin=None,
+    coords=None,
+    dims=None,
+    pred_dims=None,
+    extra_event_dims=None,
+    model=None,
+    num_samples: int = 1000,
+):
+    """Convert NumPyro SVI results into a DataTree object.
+
+    If no dims are provided, this will infer batch dim names from NumPyro model plates.
+    For event dim names, such as with the ZeroSumNormal, `infer={"event_dims":dim_names}`
+    can be provided in numpyro.sample, i.e.::
+
+        # equivalent to dims entry, {"gamma": ["groups"]}
+        gamma = numpyro.sample(
+            "gamma",
+            dist.ZeroSumNormal(1, event_shape=(n_groups,)),
+            infer={"event_dims":["groups"]}
+        )
+
+    There is also an additional `extra_event_dims` input to cover any edge cases, for instance
+    deterministic sites with event dims (which dont have an `infer` argument to provide metadata).
+
+    For a usage example read the
+    :ref:`Creating InferenceData section on from_numpyro <creating_InferenceData>`
+
+    Parameters
+    ----------
+    svi : numpyro.infer.svi.SVI
+        Numpyro SVI instance used for fitting the model.
+    svi_result : numpyro.infer.svi.SVIRunResult
+        SVI results from a fitted model.
+    model_args : tuple, optional
+        Model arguments, should match those used for fitting the model.
+    model_kwargs : dict, optional
+        Model keyword arguments, should match those used for fitting the model.\
+    prior : dict, optional
+        Prior samples from a NumPyro model
+    posterior_predictive : dict, optional
+        Posterior predictive samples for the posterior
+    predictions : dict, optional
+        Out of sample predictions
+    constant_data : dict, optional
+        Dictionary containing constant data variables mapped to their values.
+    predictions_constant_data : dict, optional
+        Constant data used for out-of-sample predictions.
+    index_origin : int, optional
+    coords : dict, optional
+        Map of dimensions to coordinates
+    dims : dict of {str : list of str}, optional
+        Map variable names to their coordinates. Will be inferred if they are not provided.
+    pred_dims : dict, optional
+        Dims for predictions data. Map variable names to their coordinates. Default behavior is to
+        infer dims if this is not provided
+    extra_event_dims : dict, optional
+        Extra event dims for deterministic sites. Maps event dims that couldnt be inferred to
+        their coordinates.
+    num_chains : int, default 1
+        Number of chains used for sampling. Ignored if posterior is present.
+
+    Returns
+    -------
+    DataTree
+    """
+    posterior = SVIWrapper(
+        svi,
+        svi_result=svi_result,
+        model_args=model_args,
+        model_kwargs=model_kwargs,
+        num_samples=num_samples,
+    )
+    with rc_context(rc={"data.sample_dims": ["samples"]}):
```

Comment:
How does this fit in with hdi (and possibly other functions) expecting (chain, draw) in future versions

Reply:
looks like arviz_stts.eti validates based on the sample_dims now so this should be fine!